exec.js

```diff
@@ -109,11 +138,42 @@ var toggleOverlay = function() {
     window.scrollTo(0,0);
 };
 
-var formValid = function() {
-    var argForm = document.getElementById("arg-form");
-    for(var i = 0; i < argForm.length; i++) {
-        if (!argForm[i].validity.valid)
-            return false;
+var formValid = function(state) {
+    var selectedIndex = state.run.get("selected");
+    var elements = document.getElementsByClassName("argument-container");
+
+    for(var i = 0; i < elements.length; i++) {
+        var textElement = elements[i].getElementsByTagName("input")[0];
+        if (textElement.type == "text") {
+            var textName = getRNameFromElement(textElement);
+            var textIndex = getIndexFromArgName(textName, state);
+            var regexType = state.run.get("plugins[" + selectedIndex +
+                            "].args[" + textIndex + "].regex");
+            var regex = placeholderToRegex(regexType);
+            if (regex.test(textElement.value) == false) 
+                return false;
+        }
+    }
+
+    return true
+}
+
+var placeholderToRegex = function(input) {
+    switch (input) {
+        case "email":
+            return /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i
+            break;
+        case "IP":
+            return /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/
+            break;
+        case "domain":
+            return /^(?!:\/\/)([a-zA-Z0-9]+\.)?[a-zA-Z0-9][a-zA-Z0-9-]+\.[a-zA-Z]{2,6}?$/i
+            break;
+        case "hostname":
+            return /(^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|(^(?!:\/\/)([a-zA-Z0-9]+\.)?[a-zA-Z0-9][a-zA-Z0-9-]+\.[a-zA-Z]{2,6}?$)/i
+            break;
+        case "URL":
+            return /[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/i
+            break;
     }
```

Comment:
What should we return from here in case `input` is different from all the cases anticipated in the `switch`?